adapters/rubicon/rubicon.go

```diff
@@ -473,3 +503,174 @@ func NewRubiconAdapter(config *adapters.HTTPAdapterConfig, uri string, xuser str
 		XAPIPassword: xpass,
 	}
 }
+
+func (a *RubiconAdapter) MakeRequests(request *openrtb.BidRequest) ([]*adapters.RequestData, []error) {
+	numRequests := len(request.Imp)
+	errs := make([]error, 0, len(request.Imp))
+	var err error
+
+	requestData := make([]*adapters.RequestData, 0, numRequests)
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	headers.Add("User-Agent", "prebid-server/1.0")
+
+	for i := 0; i < numRequests; i++ {
+		request.Imp = request.Imp[i : i+1]
```

Comment:
It seems to me like you're making a slice of len 1, and then calling `request.Imp[0]` throughout the rest of this loop.
 
For my education... what's the benefit of doing this rather than something like `thisImp := request.Imp[i]` or `for _, thisImp := range request.Imp`? Both of those seem clearer to me.

Reply:
I'm grabbing the first element from the imp array and re-creating the request.Imp object with just that first element. I guess another way to do this would be to create a copy of the imp object, make necessary changes then assign it back to request.Imp at the end.

Reply:
oh right, I see... you `reqJSON, err := json.Marshal(request)` at the end, so you want `request.Imp` to have a single element.

Don't worry about this comment, then. There's some tradeoff between simplicity and performance (making copies vs. different amounts of mutability), and it's your adapter, so your decision :).

Reply:
That said... I was confused as to why this code was working, so I took it to the playground. I don't think it's doing what you think it's doing:

https://play.golang.org/p/rXAmOTezcw

Reply:
oh yes, I see what you mean. I should have inspected this functionality instead of copying the code from our legacy endpoint. that said, this was a bug there too so I will update that. didn't know playground existed, awesome tool.